oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/resourceserver/DefaultResourceServerConfig.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.resourceserver;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.ProviderManager;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.oauth2.resourceserver.web.BearerTokenAuthenticationFilter;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import java.util.Arrays;
+
+
+/**
+ * Configuration that adds the appropriate default configuration for an
+ * OAuth2 Resource Server.
+ *
+ * May replace {@link WebSecurityConfigurerAdapter} in tests that want this pre-configuration.
+ *
+ * If further configuration of {@link HttpSecurity} is desired, then override as normal, though
+ * make sure to call super:
+ *
+ * <pre>
+ * @Override
+ * protected void configure(HttpSecurity http) throws Exception {
+ *     super.configure(http);
+ *
+ *     // .. extra http configuration
+ * }
+ * </pre>
+ *
+ * @author Josh Cummings
+ */
+public abstract class DefaultResourceServerConfig extends WebSecurityConfigurerAdapter {
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		http
+				.addFilterAfter(
+						bearerTokenAuthenticationFilter(),
+						BasicAuthenticationFilter.class)
+				.exceptionHandling().and()
+				.csrf().disable();
+	}
+
+	protected abstract AuthenticationProvider oauth2AuthenticationProvider();
+
+	public BearerTokenAuthenticationFilter bearerTokenAuthenticationFilter() {
+		return new BearerTokenAuthenticationFilter(authenticationManager());
+	}
+
+	@Bean
+	@Override
+	public AuthenticationManager authenticationManager() {
```

Comment:
I don't think you need to expose `AuthenticationManager`. In the extended `SecurityConfig`'s you can `httpSecurity.authenticationProvider()`. I think that should do it.

Reply:
Okay, I'll try that.